src/main/java/com/hms_networks/americas/sc/extensions/util/RawNumberValueUtils.java

```diff
@@ -0,0 +1,313 @@
+              + value1.getClass().getName()
+              + " and "
+              + value2.getClass().getName());
+    }
+
+    // Return min value
+    return minValue;
+  }
+
+  /**
+   * Get and return the maximum value of the two specified values.
+   *
+   * @param value1 first value
+   * @param value2 second value
+   * @return maximum value
+   * @throws IllegalArgumentException if values are not of the same type
+   * @since 1.15.7
+   */
+  public static Object getValueMax(Object value1, Object value2) {
+    // Compare values by type
+    Object maxValue;
+    if (value1 instanceof Integer && value2 instanceof Integer) {
+      // Compare values
+      Integer int1 = (Integer) value1;
+      Integer int2 = (Integer) value2;
+      if (int1.intValue() > int2.intValue()) {
+        maxValue = int1;
+      } else {
+        maxValue = int2;
+      }
+    } else if (value1 instanceof Long && value2 instanceof Long) {
+      // Compare values
+      Long long1 = (Long) value1;
+      Long long2 = (Long) value2;
+      if (long1.longValue() > long2.longValue()) {
+        maxValue = long1;
+      } else {
+        maxValue = long2;
+      }
+    } else if (value1 instanceof Float && value2 instanceof Float) {
+      // Compare values
+      Float float1 = (Float) value1;
+      Float float2 = (Float) value2;
+      if (float1.floatValue() > float2.floatValue()) {
+        maxValue = float1;
+      } else {
+        maxValue = float2;
+      }
+    } else if (value1 instanceof Double && value2 instanceof Double) {
+      // Compare values
+      Double double1 = (Double) value1;
+      Double double2 = (Double) value2;
+      if (double1.doubleValue() > double2.doubleValue()) {
+        maxValue = double1;
+      } else {
+        maxValue = double2;
+      }
+    } else if (value1 instanceof Short && value2 instanceof Short) {
+      // Compare values
+      Short short1 = (Short) value1;
+      Short short2 = (Short) value2;
+      if (short1.shortValue() > short2.shortValue()) {
+        maxValue = short1;
+      } else {
+        maxValue = short2;
+      }
+    } else if (value1 instanceof Byte && value2 instanceof Byte) {
+      // Compare values
+      Byte byte1 = (Byte) value1;
+      Byte byte2 = (Byte) value2;
+      if (byte1.byteValue() > byte2.byteValue()) {
+        maxValue = byte1;
+      } else {
+        maxValue = byte2;
+      }
+    } else {
+      // Throw exception
+      throw new IllegalArgumentException(
+          "Cannot compare values of type "
+              + value1.getClass().getName()
+              + " and "
+              + value2.getClass().getName());
+    }
+
+    // Return max value
+    return maxValue;
+  }
+
+  /**
+   * Get and return the sum of the specified values.
+   *
+   * @param values values to sum
+   * @return sum of values
+   * @throws IllegalArgumentException if values are not of the same type
+   * @since 1.15.7
+   */
+  public static Object getValueAvg(Object[] values) {
```

Comment:
This implementation does not work well for Bytes. If byte[] is longer than 255 or sum > 255

Reply:
This is a side effect of supporting all of the primitive types in Java, and this class would likely not be necessary on Java 1.5+.

It doesn't make sense for us to add special handling based on specific types. Ultimately, this is a helper method that 'replaces' newer Java code like this:

```
int sum = 0;
for (int x: values) sum += x;
```

Where `sum += x` is the same as `sum = sum + x;`, the same issue could arise here regardless of type. It is a general consideration when summing numbers, and our implementation should handle this exactly the same as using a loop to sum the values in newer Java, without a utility class.

Reply:
Java actually does do promotion from 'byte' or 'short' to an 'int' when performing operations on the values. I have updated all applicable methods/code to account for this. The JUnit test class also verifies this behavior.